src/impute.jl

```diff
@@ -117,15 +126,20 @@ julia> [k => parent(parent(v)) for (k, v) in Impute.filter(ds; dims=:loc).data]
  (:predict, :load) => [7.0; 8.1; 9.0]
 ```
 """
-Impute.apply(ds::KeyedDataset, f::Filter; dims) = Impute.apply!(deepcopy(ds), f; dims=dims)
+function Impute.apply(ds::KeyedDataset, f::Filter; dims, pattern=Pattern(:__, dims))
```

Comment:
This feels like this is an ugly interface, maybe we should just let `dims` be a `Symbol` or `Pattern` and figure out which to use internally?